### PR TITLE
Add GA4 support

### PR DESCRIPTION
**Pull request by ViliusS**
Initial GA4 support. Still working on some stuff.

### CODE REVIEW

File: ganalytics.php [hunk 1]
```diff
@@ -117,25 +107,25 @@ function gaOptout() {
...
+        } else {
+            $settings = [];
+        }
 
-        if (!$this->config->get('plugins.ganalytics.debugTrace', false)) unset ($settings['trace-debug']);
-        if (!$this->config->get('plugins.ganalytics.anonymizeIp', false)) unset ($settings['anonymize']);
-        if (!$this->config->get('plugins.ganalytics.forceSsl', false)) unset ($settings['force-ssl']);
+        $settings = array_merge($settings, $this->getCookieConfiguration());
+        if (!empty($settings)) {
+            $settings = str_replace('"', '\'', json_encode($settings));
```
Comment (Karmalakas) [inline ganalytics.php]: **Question:** Why is this replacement needed? Doesn't Google handle the default JSON escaping?

Comment (Karmalakas) [inline ganalytics.php]: **Suggestion:** It's better to wrap this into a `try-catch` and use `JSON_THROW_ON_ERROR` flag IMO

Comment (ViliusS) [reply ganalytics.php]: Fixed.

Comment (ViliusS) [reply ganalytics.php]: It was more a cosmetic fix really, but when testing various invalid JSON strings I have found that it messes with validity of the final output. Removed now.

**Comment by utagawal** [comment]
Any chance to get it fixed & merged ?

**Comment by escopecz** [comment]
This is just a draft so I expect this is work in progress. Once this PR is read and will have a couple of approvals that it's well tested and ready to go then I'll merge it and release a new version.

**Comment by ViliusS** [comment]
@utagawal you can try the latest code from this PR branch. It should fully work with one exception - some old setting values get reset because of this Grav bug https://github.com/getgrav/grav/issues/3697.
Sadly there is no feedback on Grav side on that bug, so this PR is also stuck, unless someone finds a better way to transfer those settings.

**Comment by ViliusS** [comment]
@escopecz PR is finished, tested in production. Sadly mentioned Grav bug still doesn't allow to migrate deprecated settings automatically, but since Universal Analytics cut-off date is just around the corner I think you can release v2.0. I have added a note to README that one needs to migrate deprecated settings manually after the upgrade.

**Review by escopecz — APPROVED** [review]
Makes total sense! Thanks for working on this!

I have no time to test this as I don't use this plugin anymore. But I don't want to block it. So any potential bugs will have to be fixed afterwards.

**Comment by escopecz** [comment]
@ViliusS I've also added you as a collaborator to this repo so feel free to merge other PRs and make a release.

**Pull request merged**